app/cruds/cruds_raffle.py

```diff
@@ -0,0 +1,406 @@
+"""File defining the functions called by the endpoints, making queries to the table using the models"""
+
+import logging
+
+from sqlalchemy import delete, select, update
+from sqlalchemy.exc import IntegrityError
+from sqlalchemy.ext.asyncio import AsyncSession
+from sqlalchemy.orm import selectinload
+
+from app.models import models_raffle
+from app.schemas import schemas_raffle
+
+# import random
+
+
+hyperion_error_logger = logging.getLogger("hyperion_error")
+
+
+# Raffles
+
+
+async def get_raffles(db: AsyncSession) -> list[models_raffle.Raffle]:
+    """Return all raffle from database"""
+
+    result = await db.execute(select(models_raffle.Raffle))
+    return result.scalars().all()
+
+
+async def create_raffle(
+    raffle: models_raffle.Raffle,
+    db: AsyncSession,
+) -> models_raffle.Raffle:
+    """Create a new raffle in database and return it"""
+
+    db.add(raffle)
+    try:
+        await db.commit()
+        return raffle
+    except IntegrityError:
+        await db.rollback()
+        raise ValueError("This name is already used")
+
```

Comment:
Reusing the raised error will make them easier to catch
```suggestion
    except IntegrityError as err:
        await db.rollback()
        raise ValueError(err)

```